### PR TITLE
fix: remove white spaces from subject entries before committing

### PR DESCRIPTION
**Pull request by DragosIorgulescu**

### CODE REVIEW

File: spec/requests/stash_datacite/subjects_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      let(:params_hash) do
+        {
+          resource_id: @resource.id,
+          subject: ' foo, , bar ,baz '
+        }
+      end
+
+      it 'strips subject strings and removes blanks' do
+        # when
+        post '/stash_datacite/subjects/create', params: params_hash, headers: default_authenticated_headers, as: :json
```
[the remark this answers is not in the document]
Comment (ryscher) [reply spec/requests/stash_datacite/subjects_controller_spec.rb:27]: Normally, we only use the spec/requests to test API calls. You're calling a Rails route here, so I don't know if we have equivalent authentication paths.

Maybe this test would be better as a feature test, using the UI components?

**Comment by ahamelers** [comment]
@DragosIorgulescu running `rubocop -a` will fix most ruby linting issues automatically

**Review by ahamelers — CHANGES_REQUESTED** [review]
Please run rubocop and make sure it passes!

**Comment by ryscher** [comment]
Looks good and tests well. We can figure out the authentication issue with the rspec later.

**Pull request merged**